comms/core/src/protocol/messaging/outbound.rs

```diff
@@ -104,7 +104,7 @@ impl OutboundMessaging {
                 },
                 Err(err) => {
                     metrics::error_count(&peer_node_id).inc();
-                    error!(
+                    info!(
```

Comment:
I couldn't find a good way to do this, which is why it's an error in all cases. 

There are some `io::Error`s that we want to be an error (most of the errors in `MessagingProtocolError`) and some that we don't (e.g. the remote closed the connection), but the `MessagingProtocolError::Io` case is where we want to exclude some errors.

I suggest adding a ConnectionClosed variant to MessagingProtocolError, and manually implement the conversion

```rust
impl From<io::Error> for MessagingProtocolError {
    fn from(err: Error) -> Self {
        match err.kind() {
            ErrorKind::ConnectionRefused |
            ErrorKind::ConnectionReset |
            ErrorKind::ConnectionAborted |
            ErrorKind::BrokenPipe |
            ErrorKind::UnexpectedEof => MessagingProtocolError::ConnectionClosed(err),
            _ => MessagingProtocolError::Io(err),
        }
    }
}
```

Then handle that variant here 
```rust
Err( MessagingProtocolError::ConnectionClosed(err) => {
 debug!(
                                target: LOG_TARGET,
                                "Connection closed {}: {} {}",
                                peer_node_id,
                                err.kind(),
                                err
                            );
}
```